source/usb/webusb/usbd_core_webusb.c

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file    usbd_core_webusb.c
+ * @brief   WebUSB Device driver
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2009-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "string.h"
+
+#include "RTL.h"
+#include "rl_usb.h"
+#include "usb_for_lib.h"
+
+/*
+ *  USB Device Endpoint 0 Event Callback - WebUSB specific handling (Setup Request To Device)
+ *    Parameters:      none
+ *    Return Value:    TRUE - Setup vendor request ok, FALSE - Setup vendor request not supported
+ */
+
+__weak BOOL USBD_EndPoint0_Setup_WebUSB_ReqToDevice(void)
+{
+    U8  *pD;
+    U32 len, n;
+
+    BOOL success = (__FALSE);
+    if (USBD_SetupPacket.bRequest == usbd_webusb_vendor_code) {			/* vendor code correct? */
+        switch (USBD_SetupPacket.wIndex) {
+            case WEBUSB_REQUEST_GET_ALLOWED_ORIGINS:
+                pD = (U8 *)USBD_WebUSBAllowedOriginsHeader;
+                USBD_EP0Data.pData = pD;
+                len = ((WEBUSB_ALLOWED_ORIGINS_HEADER_DESCRIPTOR *)pD)->wTotalLength;
+                success = (__TRUE);
+                break;
+
+            case WEBUSB_REQUEST_GET_URL:
+                pD = (U8 *)USBD_WebUSBURLDescriptor;
+                if (USBD_SetupPacket.wValueL == 0) {
+                    success = (__FALSE);
+                    break;
+                }
+
+                for (n = 0; n + 1 < USBD_SetupPacket.wValueL; n++) {
+                    if (((WEBUSB_URL_DESCRIPTOR *)pD)->bLength != 0) {
+                        pD += ((WEBUSB_URL_DESCRIPTOR *)pD)->bLength;
+                    }
+                }
+
+                if (((WEBUSB_URL_DESCRIPTOR *)pD)->bLength == 0) {
+                    success = (__FALSE);
+                    break;
+                }
+
```

Comment:
info_get_board_id() should be called here to send board_id to redirection service. Will any other field be useful in the future? Should all fields in `Details.txt` field be sent?

Example:
```
# DAPLink Firmware - see https://mbed.com/daplink
Unique ID: 0240000032044e45003a700a997b00016781000097969900
HIC ID: 97969900
Auto Reset: 0
Automation allowed: 0
Overflow detection: 0
Daplink Mode: Interface
Interface Version: 0244
Git SHA: c72963241000edb2e913ea806e62e98c2cd11552
Local Mods: 0
USB Interfaces: MSD, CDC, HID
Interface CRC: 0x9ec355e2
Remount count: 0
```